packages/core/src/components/menu-category/test/menu-category.ct.ts

```diff
@@ -320,3 +320,59 @@ test('show category if item are focused', async ({ mount, page }) => {
   const dropdown = categoryElement.locator('ix-dropdown');
   await expect(dropdown).toBeVisible();
 });
+
+test('should adjust height when items are added dynamically', async ({
+  mount,
+  page,
+}) => {
+  await mount(`
+    <ix-application>
+      <ix-menu>
+        <ix-menu-category label="Category label">
+          <ix-menu-item>Test</ix-menu-item>
+          <ix-menu-item>Test</ix-menu-item>
+        </ix-menu-category>
+      </ix-menu>
+    </ix-application>
+  `);
+
+  await page.locator('ix-menu-expand-icon').click();
+  await page.locator('ix-menu-category').click();
```

Comment:
This way you don't have to call locator/querySelector multiple times
```suggestion
  const menuCategory = page.locator('ix-menu-category');
  await menuCategory.click();
```